src/commands/init/ai-rules.ts

```diff
@@ -0,0 +1,447 @@
+import { resolve } from 'node:path'
+import { promises as fs } from 'node:fs'
+import type { NetlifyAPI } from '@netlify/api'
+
+import { chalk, log, logAndThrowError, type APIError } from '../../utils/command-helpers.js'
+import { normalizeRepoUrl } from '../../utils/normalize-repo-url.js'
+import { runGit } from '../../utils/run-git.js'
+import { startSpinner } from '../../lib/spinner.js'
+import { getContextConsumers, type ConsumerConfig } from '../../recipes/ai-context/context.js'
+import execa from '../../utils/execa.js'
+import { version } from '../../utils/command-helpers.js'
+import type BaseCommand from '../base-command.js'
+import type { SiteInfo } from '../../utils/types.js'
+import inquirer from 'inquirer'
+
+interface ProjectInfo {
+  success: boolean
+  projectId: string
+  prompt: string
+}
+
+// Check if a command belongs to a known IDE (reusing ai-context logic)
+const getConsumerKeyFromCommand = (command: string, consumers: ConsumerConfig[]): string | null => {
+  const match = consumers.find(
+    (consumer) => consumer.consumerProcessCmd && command.includes(consumer.consumerProcessCmd),
+  )
+  return match ? match.key : null
+}
+
+// Get command and parent PID (same logic as ai-context)
+const getCommandAndParentPID = async (
+  pid: number,
+): Promise<{
+  parentPID: number
+  command: string
+  consumerKey: string | null
+}> => {
+  const { stdout } = await execa('ps', ['-p', String(pid), '-o', 'ppid=,comm='])
+  const output = stdout.trim()
+  const spaceIndex = output.indexOf(' ')
+  const parentPID = output.substring(0, spaceIndex)
+  const command = output.substring(spaceIndex + 1).toLowerCase()
+
+  const consumers = await getContextConsumers(version) // Use current CLI version
+
+  return {
+    parentPID: Number(parentPID),
+    command,
+    consumerKey: getConsumerKeyFromCommand(command, consumers),
+  }
+}
+
+// Detect IDE by walking up process tree (same logic as ai-context)
+const detectIDE = async (): Promise<ConsumerConfig | null> => {
+  if (process.env.AI_CONTEXT_SKIP_DETECTION === 'true') {
+    return null
+  }
+
+  const ppid = process.ppid
+  let result: Awaited<ReturnType<typeof getCommandAndParentPID>>
+  try {
+    result = await getCommandAndParentPID(ppid)
+    while (result.parentPID !== 1 && !result.consumerKey) {
+      result = await getCommandAndParentPID(result.parentPID)
+    }
+  } catch {
+    // Process detection failed
+    return null
+  }
+
+  if (result.consumerKey) {
+    const consumers = await getContextConsumers(version)
+    const contextConsumer = consumers.find((consumer) => consumer.key === result.consumerKey)
+    if (contextConsumer) {
+      return contextConsumer
+    }
+  }
+
+  return null
+}
+
+// Generate MCP configuration for the detected IDE
+const generateMcpConfig = (ide: ConsumerConfig): string => {
```

Comment:
Why are you returning the config as a string here only to then re-parse it into an object in `configureMcpForVSCode`?